tdp/cli/commands/stale/__init__.py

```diff
@@ -0,0 +1,121 @@
+                session
+            )
+            modified_components_names = cluster_variables.get_modified_components_names(
+                services_components_versions=deployed_component_version_logs
+            )
+            stale_components = set()
+            for modified_component_name in modified_components_names:
+                click.echo(f"{modified_component_name.full_name} has been modified.")
+                if modified_component_name.is_service:
+                    service_operations = filter(
+                        lambda operation: operation.action_name == "config"
+                        and not operation.is_service_operation(),
+                        dag.services_operations[modified_component_name.service_name],
+                    )
+                    stale_components.update(
+                        [
+                            StaleComponent(
+                                service_name=service_operation.service_name,
+                                component_name=service_operation.component_name,
+                                to_reconfigure=True,
+                                to_restart=True,
+                            )
+                            for service_operation in service_operations
+                        ]
+                    )
+                else:
+                    stale_components.add(
+                        StaleComponent(
+                            service_name=modified_component_name.service_name,
+                            component_name=modified_component_name.component_name,
+                            to_reconfigure=True,
+                            to_restart=True,
+                        )
+                    )
+            session.query(StaleComponent).delete()
+            session.add_all(stale_components)
+            session.commit()
+        # Print stale components
+        stale_components = session.query(StaleComponent).all()
+        _print_stale_components(stale_components)
+
+
+def _print_stale_components(stale_components: List[StaleComponent]):
+    """Print the list of stale components.
+
+    Args:
+        stale_components: The list of stale components to print.
+    """
+    headers = StaleComponent.__table__.columns.keys()
+    click.echo(
+        tabulate(
+            [
+                _format_stale_component(stale_component, headers)
+                for stale_component in stale_components
+            ],
+            headers="keys",
+        )
+    )
+
+
+def _format_stale_component(
+    stale_component: StaleComponent, headers: List[str]
+) -> dict:
+    """Format a StaleComponent object into a dict for tabulate.
+
+    Args:
+        stale_component: The StaleComponent object to format.
+        headers: The headers to use for the tabulate table.
+
+    Returns:
+        A dict with the formatted StaleComponent object.
+    """
+
+    def custom_format(key, value):
+        if key == "to_reconfigure" or key == "to_restart":
```

Comment:
"then" and "else" are the same, I don't understand the use of "if".

Reply:
You're right, I removed this function.